src/react-integration/hooks/useResourceNew.ts

```diff
@@ -1,49 +1,36 @@
-import { ReadShape, Schema, RequestResource, SchemaOf } from '~/resource';
+import { ReadShape, Schema, SchemaOf } from '~/resource';
 import useCacheNew from './useCacheNew';
 import useRetrieve from './useRetrieve';
 import useError from './useError';
 import { useMemo } from 'react';
 
+import hasUsableData from './hasUsableData';
+
 type ResourceArgs<
   S extends Schema,
   Params extends Readonly<object>,
   Body extends Readonly<object | string> | void
 > = [ReadShape<S, Params, Body>, Params | null];
 
-/** If the invalidIfStale option is set we suspend if resource has expired */
-function hasUsableData<
-  S extends Schema,
-  Params extends Readonly<object>,
-  Body extends Readonly<object | string> | void
->(
-  resource: RequestResource<ReadShape<S, Params, Body>> | null,
-  fetchShape: ReadShape<S, Params, Body>,
-) {
-  return !(
-    (fetchShape.options && fetchShape.options.invalidIfStale) ||
-    !resource
-  );
-}
-
```

Comment:
useResourceNew? I'm confused

Reply:
Did you read the rest hooks update doc? also: https://github.com/coinbase/rest-hooks/commit/53c1922f2aa6fcc9b9dbf0889aebf78516581a85